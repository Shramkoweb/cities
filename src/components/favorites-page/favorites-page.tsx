import * as React from "react";
import {Link} from 'react-router-dom';
import {connect} from "react-redux";

import PageLayout from "../page-layout/page-layout";
import {Operation} from "../../reducer/data/data";
import {getGroupingFavoritesByCities} from "../../reducer/data/selector";
import FavoritesList from "../favorites-list/favorites-list";
import Header from "../header/header";

interface FavoritesPageProps {
  favorites: any,
  onLoadFavorites: any,
  cities: any,
}

class FavoritesPage extends React.PureComponent<FavoritesPageProps> {
  componentDidMount() {
    const {onLoadFavorites} = this.props;

    onLoadFavorites();
  }

  render() {
    const {favorites, cities} = this.props;

    return (
      <PageLayout pageClasses={cities.length === 0 ? [`page--favorites-empty`] : [``]}>
        <Header/>
        {
          cities.length !== 0 ?
            <main className="page__main page__main--favorites">
              <div className="page__favorites-container container">
                <section className="favorites">
                  <h1 className="favorites__title">Saved listing</h1>
                  <ul className="favorites__list">
                    <FavoritesList favorites={favorites} cities={cities}/>
                  </ul>
                </section>
              </div>
            </main>
            : <main className="page__main page__main--favorites page__main--favorites-empty">
              <div className="page__favorites-container container">
                <section className="favorites favorites--empty">
                  <h1 className="visually-hidden">Favorites (empty)</h1>
                  <div className="favorites__status-wrapper">
                    <b className="favorites__status">Nothing yet saved.</b>
                    <p className="favorites__status-description">Save properties to narrow down search or plan yor future
                      trips.</p>
                  </div>
                </section>
              </div>
            </main>
        }
        <footer className="footer container">
          <Link to="/" className="footer__logo-link">
            <img className="footer__logo" src="img/logo.svg" alt="6 cities logo" width="64" height="33"/>
          </Link>
        </footer>
      </PageLayout>
    );
  }
}


const mapStateToProps = (state) => {
  const favorites = getGroupingFavoritesByCities(state);

  return {
    cities: Object.keys(favorites),
    favorites: favorites || {},
  };
};

const mapDispatchToProps = (dispatch) => ({
  onLoadFavorites: () => dispatch(Operation.loadFavorites()),
});

export {FavoritesPage};
export default connect(mapStateToProps, mapDispatchToProps)(FavoritesPage);
