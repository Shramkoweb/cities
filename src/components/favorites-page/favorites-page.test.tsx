import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import {configure, shallow} from "enzyme";
import {FavoritesPage} from "./favorites-page";

configure({adapter: new Adapter()});

jest.mock(`../page-layout/page-layout.tsx`, () => jest.fn().mockReturnValue(null));
jest.mock(`../place-card/place-card.tsx`, () => jest.fn().mockReturnValue(null));
jest.mock(`../header/header.tsx`, () => jest.fn().mockReturnValue(null));

describe(`FavoritesPage should correctly render`, () => {
  it(`FavoritesPage is renders correctly`, () => {
    const favorites = shallow(
      <FavoritesPage
        cities={[`Amsterdam`]}
        favorites={{}}
        onLoadFavorites={jest.fn()}
      />
    );

    expect(favorites).toMatchSnapshot();
  });

  it(`FavoritesPage empty page renders correctly`, () => {
    const favorites = shallow(
      <FavoritesPage
        cities={[]}
        favorites={{}}
        onLoadFavorites={jest.fn()}
      />
    );

    expect(favorites).toMatchSnapshot();
  });
});
