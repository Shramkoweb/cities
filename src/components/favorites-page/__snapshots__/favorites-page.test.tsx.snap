// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesPage should correctly render FavoritesPage empty page renders correctly 1`] = `ShallowWrapper {}`;

exports[`FavoritesPage should correctly render FavoritesPage is renders correctly 1`] = `ShallowWrapper {}`;
